src/travel-aid/TravelAidScreenComponent.tsx

```diff
@@ -65,47 +62,77 @@ export const TravelAidScreenComponent = ({
         type="medium"
       />
       <ScrollView contentContainerStyle={styles.container}>
-        {state && (
-          <Section>
-            <GenericSectionItem style={styles.sectionContainer}>
-              <EstimatedCallInfo
-                departure={{
-                  cancellation: false,
-                  predictionInaccurate: false,
-                  serviceJourney: serviceJourney ?? {},
-                  destinationDisplay:
-                    state.focusedEstimatedCall.destinationDisplay,
-                }}
-              />
-            </GenericSectionItem>
-            <GenericSectionItem>
-              <View style={styles.sectionContainer}>
-                {state.status === TravelAidStatus.NoRealtime && (
-                  <MessageInfoBox
-                    type="error"
-                    title={t(TravelAidTexts.noRealtimeError.title)}
-                    message={t(TravelAidTexts.noRealtimeError.message)}
-                  />
-                )}
-                <View style={styles.subContainer}>
-                  <ThemeText type="body__tertiary--bold">
-                    {getStopHeader(state.status, t)}
-                  </ThemeText>
-                  <ThemeText type="heading__title">
-                    {state.focusedEstimatedCall.quay.stopPlace?.name}{' '}
-                    {state.focusedEstimatedCall.quay.publicCode}
-                  </ThemeText>
-                </View>
-                <TimeInfo state={state} />
-              </View>
-            </GenericSectionItem>
-          </Section>
+        {isLoading && <ActivityIndicator size="large" />}
+        {(isError || (!isLoading && !serviceJourney?.estimatedCalls)) && (
+          <MessageInfoBox
+            type="error"
+            message={t(TravelAidTexts.error.message)}
+            onPressConfig={{action: refetch, text: t(dictionary.retry)}}
+          />
+        )}
+        {serviceJourney?.estimatedCalls && (
+          <TravelAidSection
+            serviceJourney={serviceJourney}
+            fromQuayId={serviceJourneyDeparture.fromQuayId}
+          />
```

Comment:
There is some stuff here that seems unnatural or suboptimal with regards to usage of the react-query status and data. We can have a quick look at it together.